components/util/thread.rs

```diff
@@ -62,26 +76,30 @@ pub trait SendOnFailure {
 impl<T> SendOnFailure for Sender<T> where T: Send + 'static {
     type Value = T;
     fn send_on_failure(&mut self, value: T) {
-        self.send(value).unwrap();
+        // Discard any errors to avoid double-panic
+        let _ = self.send(value);
     }
 }
 
 impl<T> SendOnFailure for IpcSender<T> where T: Send + Serialize + 'static {
     type Value = T;
     fn send_on_failure(&mut self, value: T) {
-        self.send(value).unwrap();
+        // Discard any errors to avoid double-panic
+        let _ = self.send(value);
```

Comment:
I'm a bit worried about even putting a warn! here. Anything that panics in a panic handler takes the entire process down with it. If the send fails, chances are the entire constellation is shutting down.